cmd/simulacra/simulacra.go

```diff
@@ -187,52 +201,76 @@ func getConfigFromYaml(configPath string) (Config, error) {
 		config.Name = getInstanceName()
 	}
 
-	return config, nil
+	return &config, nil
 }
 
-func getSimulacraConfig() (Config, error) {
+func getSimulacraConfig() (*Config, error) {
 	configPath := flag.String("config", "", "Optional. The path to a YAML file specifying all the configurations for Simulacra. If unspecified, Simulacra will either use values from other command line arguments or use default values. If specifed along with other command line arguments, all others will be ignored.")
 	platform := flag.String("platform", defaultPlatform, "Optional. The OS for the VM. If missing, debian-11 is used.")
 	opsAgentConfigFile := flag.String("ops_agent_config", "", "Optional. Path to the Ops Agent Config File. If unspecified, Ops Agent will not install any third party applications and configure Ops Agent with default settings. ")
 	project := flag.String("project", "", "Optional. If missing, Simulacra will try to infer from GCloud config.")
 	zone := flag.String("zone", "", "Optional. If missing, Simulacra will try to infer from GCloud config. ")
 	name := flag.String("name", getInstanceName(), "Optional. A name for the instance to be created. If missing, a random name with prefix 'simulacra-vm-instance' will be assigned. ")
 	thirdPartyAppsPath := flag.String("third_party_apps_path", defaultThirdPartyAppsPath, "Optional. The path to the third party apps data folder. If missing, Simulacra assumes the working directory is the root of the repo. Therefore, the default path is './integration_test/third_party_apps_data' ")
+	serviceAccount := flag.String("service_account", "", "Optional. A service account for the VM. If missing, the VM will be instantiated with a default service account.")
 	flag.Parse()
 
 	if *configPath != "" {
 		return getConfigFromYaml(*configPath)
 	}
 
-	config := Config{Platform: *platform, ConfigFilePath: *opsAgentConfigFile, Project: *project, Zone: *zone, Name: *name,
-		ThirdPartyAppsPath: *thirdPartyAppsPath}
+	config := Config{
+		Platform:           *platform,
+		ConfigFilePath:     *opsAgentConfigFile,
+		Project:            *project,
+		Zone:               *zone,
+		Name:               *name,
+		ThirdPartyAppsPath: *thirdPartyAppsPath,
+		ServiceAccount:     *serviceAccount,
+	}
 
-	return config, nil
+	return &config, nil
 
 }
 
-func runCustomScripts(ctx context.Context, vm *gce.VM, logger *logging.DirectoryLogger, scripts []string) error {
-	if len(scripts) == 0 {
-		log.Default().Print("No Custom Scripts To Run ")
-		return nil
-	}
+func runCustomScripts(ctx context.Context, vm *gce.VM, logger *logging.DirectoryLogger, scripts []*Script) error {
+	for _, script := range scripts {
+		scriptContent, err := os.ReadFile(script.Path)
```

Comment:
change based on : https://github.com/GoogleCloudPlatform/ops-agent/pull/1326 , the goal was to remove the else